### DISCUSSION TITLE
Clicking Buttons in a React Overlay

### DISCUSSION THREAD

**Discussion by FlowFlaux** [seleniumbase/SeleniumBase#2648] [Q&A] [answered]
I got a page with buttons that appear dynamically upon certain changes.
Clicking one of those buttons loads up a react overlay that has a 'confirm' button `<button type="button">Confirm</button>`
that I need to click, but no matter which selector, Xpath or click method I use, SB tells me that the element is not visible.
Locating it with sb.get_element("//button[text()='Confirm']") returns an element, but adding a .click() to it doesn't work. It says it's not interactable, even though I can click it with my mouse.
sb.is_text_visible() is able to see text present on that overlay, which tells me that Selenium can see it, but confuses me as to why it can't see the button.
sb.show_element doesn't help either.
I am considering just using pyautogui at this point, but I'd rather be able to click it with Selenium. Would appreciate any pointers.

**Comment by mdmintz** [comment, marked as answer]
There's `sb.js_click(selector)` for clicking elements that are not visible or not interactable due to another element receiving the click. (That could happen if there's an overlay that got in the way.)

**Reply by FlowFlaux** [reply]
I've tried that too, and it doesn't return an error, but it doesn't click the button either.

**Reply by mdmintz** [reply]
Try it with a valid CSS Selector for JS. Your XPath (`text()='Confirm'`) has no JS-valid CSS equivalent.

**Reply by FlowFlaux** [reply]
A-ha, that worked right away. Thanks a bunch, wouldn't have figured it out without you :)
